compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -763,6 +763,13 @@ void setForeachTypedBindingPatternType(BLangForeach foreachNode) {
                 BArrayType arrayType = (BArrayType) collectionType;
                 mapType.constraint = arrayType.eType;
                 break;
+            case TypeTags.TUPLE:
+                BTupleType tupleType = (BTupleType) collectionType;
+                LinkedHashSet<BType> tupleTypes = new LinkedHashSet<>(tupleType.tupleTypes);
+                mapType.constraint = tupleTypes.size() == 1 ?
+                        tupleTypes.iterator().next() :
```

Comment:
Do we have a test for this tupleTypes.size() = 1  case?

Reply:
I've already added one in line 82 of `foreach_tuples.bal`